include/interfaces/poisson_problem.h

```diff
@@ -0,0 +1,147 @@
+  void compute_system_operators(const DoFHandler<dim,spacedim> &,
+                                const std::vector<shared_ptr<LATrilinos::BlockMatrix> >,
+                                LinearOperator<LATrilinos::VectorType> &,
+                                LinearOperator<LATrilinos::VectorType> &) const;
+
+private:
+  mutable shared_ptr<TrilinosWrappers::PreconditionJacobi> preconditioner;
+
+
+};
+
+template <int dim, int spacedim, typename LAC>
+PoissonProblem<dim,spacedim, LAC>::
+PoissonProblem():
+  Interface<dim,spacedim,PoissonProblem<dim,spacedim,LAC>, LAC >("Poisson problem",
+      1,1,
+      "FESystem[FE_Q(1)]",
+      "u","1")
+{
+  this->init();
+}
+
+
+
+template <int dim, int spacedim, typename LAC>
+template <typename EnergyType, typename ResidualType>
+void
+PoissonProblem<dim,spacedim,LAC>::
+energies_and_residuals(const typename DoFHandler<dim,spacedim>::active_cell_iterator &cell,
+                       FEValuesCache<dim,spacedim> &fe_cache,
+                       std::vector<EnergyType> &,
+                       std::vector<std::vector<ResidualType> > &local_residuals,
+                       bool compute_only_system_terms) const
+{
+  const FEValuesExtractors::Scalar s(0);
+
+  ResidualType rt = this->alpha; // dummy number to define the type of variables
+  this->reinit (rt, cell, fe_cache);
+  auto &uts = fe_cache.get_values("solution_dot", "u", s, rt);
+  auto &gradus = fe_cache.get_gradients("solution", "u", s, rt);
+
+  const unsigned int n_q_points = uts.size();
+  auto &JxW = fe_cache.get_JxW_values();
+
+  auto &fev = fe_cache.get_current_fe_values();
+
+  for (unsigned int q=0; q<n_q_points; ++q)
+    {
+      auto &ut = uts[q];
+      auto &gradu = gradus[q];
+      for (unsigned int i=0; i<local_residuals[0].size(); ++i)
+        {
+          auto v = fev[s].value(i,q);
+          auto gradv = fev[s].gradient(i,q);
+          local_residuals[0][i] += (
+                                     ut*v
+                                     +
+                                     gradu*gradv
+                                   )*JxW[q];
+        }
+
+      (void)compute_only_system_terms;
+
+    }
+
+
+}
+
+
+template <int dim, int spacedim, typename LAC>
+void
+PoissonProblem<dim,spacedim,LAC>::compute_system_operators(const DoFHandler<dim,spacedim> &,
+                                                           const std::vector<shared_ptr<LATrilinos::BlockMatrix> > matrices,
+                                                           LinearOperator<LATrilinos::VectorType> &system_op,
+                                                           LinearOperator<LATrilinos::VectorType> &prec_op) const
+{
+
+  preconditioner.reset  (new TrilinosWrappers::PreconditionJacobi());
+  preconditioner->initialize(matrices[1]->block(0,0));
+  auto P = linear_operator<LATrilinos::VectorType::BlockType>(matrices[0]->block(0,0));
+
+  auto A  = linear_operator<LATrilinos::VectorType::BlockType>( matrices[0]->block(0,0) );
+
```

Comment:
In this example, can we avoid using a solver around the preconditioner? We already have a preconditioner... just use the preconditioner, not a solver around it...

~~~
auto P_inv = linear_operator<LATrilinos::VectorType::BlockType>(*preconditioner, matrices[0]->block(0,0));
~~~

or

~~~
auto P_inv = linear_operator<LATrilinos::VectorType::BlockType>(matrices[0]->block(0,0), *preconditioner);
~~~

I never remember the order of the exemplar VS the object...